### PR TITLE
Expand GA analytics to include timing and exception tracking

### LINKED ISSUE #2212: Project Loading Slowly GA event fires even though a project loaded quickly.
Oh boy. I was reviewing @humphd's analytics PR and logging all the GA calls and lo and behold I saw this thing pop up in the console even though my project loaded just fine (in less than 5 seconds). 

image

Tested this in Chrome and FF. Happens on both signed out and signed in projects.

### PR DESCRIPTION
**Pull request by humphd**
I notice that we can include timing and exception analytics in GA along with event categories.  I've expanded the API and added some new tracking code to help us see what's happening.

Fixes #2212

### CODE REVIEW

File: public/editor/scripts/analytics.js [hunk 1]
```diff
@@ -65,14 +70,71 @@
...
+      } else {
+        warn("Expected `value` arg to be a Number.");
+        return;
+      }
+    }
+
+    if(options.label) {
+      eventOptions.timingLabel = trim(options.label);
+    }
+  }
```
Comment (flukeout) [inline public/editor/scripts/analytics.js]: The ``timing()`` function ends without ever calling ``gaSend``.

**Comment by flukeout** [comment]
The documentation for exception shows the event structure as...

image

Yet when I console, log out the event object you're creating it's...

```
Object {
  hitType: "exception", 
  exDescription: "Unable to load Bramble editor in this browser", 
  exFatal: true
}
```

Will this alternate way work, or do we need to structure it like the docs say?

**Comment by flukeout** [comment]
Also thanks for leaving links to the docs in the comments, that's a handy practice.

**Comment by humphd** [comment]
OK, I've fixed the issues you found (you can indeed call the functions by passing the object vs. expanding the args out--both ways work).  I've also folded your timeout changes into this, and refactored our startup so we can cancel the timeout later. @gideonthomas you might want to take a look at this, too.

@flukeout can you test again, please?

**Comment by flukeout** [comment]
Testing again now.

**Comment by flukeout** [comment]
Just tested this and exceptions, timings and events seem to work well. The only difference I noticed is that you removed the ``eventLabel`` of the Black Screen troubleshooting event. We used to toString() the ``err`` and pass it along. Other than that looks good on my end.

**Comment by flukeout** [comment]
cc @humphd

File: public/editor/scripts/analytics.js [hunk 2]
```diff
@@ -65,14 +70,71 @@
...
+        return;
+      }
+      // Force value to int
+      eventOptions["timingValue"] = options.value|0;
+    } else {
+      // If now value is given, assume we want to try to measure time since page load
+      if(window.performance) {
+        eventOptions["timingValue"] = Math.round(window.performance.now());
+      } else {
+        warn("Expected `value` arg to be a Number.");
```
Comment (gideonthomas) [inline public/editor/scripts/analytics.js]: wrong message?

**Comment by gideonthomas** [comment]
This also fixes https://github.com/mozilla/thimble.mozilla.org/issues/2212

**Pull request merged**